contracts/Libraries/Validate.sol

```diff
@@ -14,16 +14,19 @@ library Validate {
         bytes memory sig2 = ByteUtils.slice(sigs, 65, 65);
         bytes memory confSig1 = ByteUtils.slice(sigs, 130, 65);
         bytes32 confirmationHash = keccak256(txHash, sig1, sig2, rootHash);
-        if (inputCount == 0) {
+        
+        if (blknum1 == 0 && blknum2 == 0) {
             return msg.sender == ECRecovery.recover(confirmationHash, confSig1);
         }
-        if (inputCount < 1000000) {
-            return ECRecovery.recover(txHash, sig1) == ECRecovery.recover(confirmationHash, confSig1);
-        } else {
+        bool check1 = true;
+        bool check2 = true;
+        if (blknum1 > 0) {
+            check1 = ECRecovery.recover(txHash, sig1) == ECRecovery.recover(confirmationHash, confSig1);
+        } 
+        if (blknum2 > 0) {
             bytes memory confSig2 = ByteUtils.slice(sigs, 195, 65);
-            bool check1 = ECRecovery.recover(txHash, sig1) == ECRecovery.recover(confirmationHash, confSig1);
-            bool check2 = ECRecovery.recover(txHash, sig2) == ECRecovery.recover(confirmationHash, confSig2);
-            return check1 && check2;
+            check2 = ECRecovery.recover(txHash, sig2) == ECRecovery.recover(confirmationHash, confSig2);
```

Comment:
same again

Reply:
Declaring and assigning bool check1 and check2 outside of the if statement is necessary. If it wasn't assigned to default to true, then single input transactions would fail as "return check1 & check2" would contain one false value even though the input may have been valid. 

If check1 and check2 are declared inside the if statement then they will be out of scope and inaccessible in the line "return check1 & check2"